Hide sections in home view when there are no items to display

insomniachi opened the pull request:
- Hide sections in home view when there are no items to display
- Reduce duplicated xaml

insomniachi committed:
- hide sections that has 0 items
- reduce duplicated xaml

dfederm commented on Jellyfin/Views/HomeViewModel.cs:
```diff
@@ -29,19 +30,38 @@ internal sealed partial class HomeViewModel : ObservableObject
 
+    [ObservableProperty]
+    private List<HomeViewSection> _sections;
+
```
Nit: remove extra newline

insomniachi replied:
removed

dfederm commented on Jellyfin/Common/EmptyCollectionToVisibilityConverter.cs:
```diff
@@ -0,0 +1,22 @@
+    public object Convert(object value, Type targetType, object parameter, string language)
+    {
+        return value is not IEnumerable<object> enumerable
+            ? Visibility.Collapsed
```
Should this just throw to prevent accidental usage with a non-collection?

insomniachi replied:
now throws if not ICollection

dfederm commented on Jellyfin/Common/EmptyCollectionToVisibilityConverter.cs:
```diff
@@ -0,0 +1,22 @@
+    {
+        return value is not IEnumerable<object> enumerable
+            ? Visibility.Collapsed
+            : enumerable.Count() == 0 ? Visibility.Collapsed : Visibility.Visible;
```
Use `Any()` instead of `Count() == 0`.

Or better yet, avoid Linq and just require that it's an `ICollection` instead of an `IEnumerable`

insomniachi replied:
changed to ICollection, but still keeping Count == 0 instead of Any() since count is pre computed.

dfederm commented on Jellyfin/Models/HomeViewSection.cs:
```diff
@@ -0,0 +1,10 @@
+public class HomeViewSection
+{
+    public string Name { get; set; }
+    public IEnumerable<BaseItemDto> Items { get; set; }
```
Related to other comment, consider something like `IReadOnlyList`?

insomniachi replied:
changed

insomniachi committed:
- replace IEnumerable with collection interfaces
- remove empty spaces
- EmptyCollectionToVisibilityConverter now throw for not collection types

insomniachi committed:
rename variable
